File: summer/WebContent/js/MS/internal/data/BindingValueChangedEventArgs.js
/**
 * BindingValueChangedEventArgs
 */

define(["dojo/_base/declare", "system/Type", "system/EventArgs"], 
		function(declare, Type, EventArgs){
	var BindingValueChangedEventArgs = declare("BindingValueChangedEventArgs", EventArgs,{
		constructor:function(/*object*/ oldValue, /*object*/ newValue) 
        {
            this._oldValue = oldValue;
            this._newValue = newValue;
        } 
	});
	
	Object.defineProperties(BindingValueChangedEventArgs.prototype,{
        /// <summary>
        /// The old value of the binding. 
        /// </summary>
//        public object 
		OldValue: 
        { 
            get:function() { return this._oldValue; }
        }, 

        /// <summary>
        /// The new value of the binding.
        /// </summary> 
//        public object 
        NewValue:
        { 
            get:function() { return this._newValue; } 
        }		  
	});
	
	BindingValueChangedEventArgs.Type = new Type("BindingValueChangedEventArgs", BindingValueChangedEventArgs,
			[EventArgs.Type]);
	return BindingValueChangedEventArgs;
});

File: org.summer.view.widget/src/org/summer/view/widget/data/BindingValueChangedEventArgs.java
package org.summer.view.widget.data;

import org.summer.view.widget.EventArgs;

/// <summary> 
/// Arguments for BindingValueChanged events.
/// </summary> 
/*internal*/ public class BindingValueChangedEventArgs extends EventArgs
{
    //-----------------------------------------------------
    // 
    //  Constructors
    // 
    //----------------------------------------------------- 

    /*internal*/ public BindingValueChangedEventArgs(Object oldValue, Object newValue) 
    {
    	super();
        _oldValue = oldValue;
        _newValue = newValue;
    } 

    //------------------------------------------------------ 
    // 
    //  Public Properties
    // 
    //-----------------------------------------------------

    /// <summary>
    /// The old value of the binding. 
    /// </summary>
    public Object OldValue 
    { 
        get { return _oldValue; }
    } 

    /// <summary>
    /// The new value of the binding.
    /// </summary> 
    public Object NewValue
    { 
        get { return _newValue; } 
    }

    //------------------------------------------------------
    //
    //  Private Fields
    // 
    //------------------------------------------------------

    private Object _oldValue, _newValue; 
}